src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,302 @@
+            saveFile();
+            break;
+        case COMMAND_DEADLINE:
+            splitDeadline(userInput);
+            performAddDeadline(deadlineDescription, by);
+            saveFile();
+            break;
+        case COMMAND_EVENT:
+            splitEvent(userInput);
+            performAddEvent(eventDescription, at);
+            saveFile();
+            break;
+        case COMMAND_DELETE:
+            handleIntConversion(userInput, COMMAND_DELETE);
+            performDelete(inputNum);
+            saveFile();
+            break;
+        case COMMAND_BYE:
+            exitProgram();
+            break;
+        default:
+            printInvalidInput();
+        }
+    }
+
+    private static void printInvalidInput() {
+        System.out.println(HORIZONTAL_LINE);
+        System.out.println(SAD_FACE + " OOPS! I'm sorry, but I don't know what that means " + SAD_FACE);
+    }
+
+    private static void performListTask() {
+        System.out.println(HORIZONTAL_LINE);
+        System.out.println("Here are the tasks in your list:");
+
+        for (Task t : taskArrayList) {
+            System.out.println(INDENT + t.taskNum + "." + t);
+        }
+    }
+
+    private static void handleIntConversion(String userInput, String command) {
+        number = userInput.replace(command, "");
+        number = number.trim();
+        inputNum = Integer.parseInt(number);
+    }
+
+    private static void performMarkTaskDone(int inputNum) throws NumberOutOfBoundsException {
+        //index of taskNum in taskArrayList
+        int doneTaskNum = inputNum - 1;
+
+        boolean isValidNum = (inputNum > 0) && (inputNum <= Task.taskCount);
+        if (isValidNum) {
+            taskArrayList.get(doneTaskNum).markAsDone();
+            System.out.println(HORIZONTAL_LINE);
+            System.out.println("Nice! I've marked this task as done:");
+            System.out.println(INDENT + taskArrayList.get(doneTaskNum).taskNum + "." + taskArrayList.get(doneTaskNum));
+        } else {
+            throw new NumberOutOfBoundsException();
+        }
+    }
+
+    private static void performDelete(int inputNum) throws NumberOutOfBoundsException {
+        //index of taskNum in taskArrayList
+        int deleteTaskNum = inputNum - 1;
+
+        boolean isValidNum = (inputNum > 0) && (inputNum <= Task.taskCount);
+        if (isValidNum) {
+            System.out.println(HORIZONTAL_LINE);
+            System.out.println("Noted. I've removed this task:");
+            System.out.println(INDENT + taskArrayList.get(deleteTaskNum).taskNum + "." + taskArrayList.get(deleteTaskNum));
+            taskArrayList.remove(deleteTaskNum);
+            updateTaskCount();
+            updateTaskNum(deleteTaskNum);
+            System.out.println("Now you have " + Task.taskCount + " tasks in the list");
+        } else {
+            throw new NumberOutOfBoundsException();
+        }
+    }
+
+    private static void updateTaskCount(){
```

Comment:
Since this method is strictly decreasing the taskCount, you may want to rename the method to decrementTaskCount()